tests/System.IO.Pipelines.Tests/ReadableBufferReaderFacts.cs

```diff
@@ -39,71 +49,117 @@ public void PeekReturnsMinuOneByteInTheEnd()
-
-                Assert.Equal(1, reader.Take());
-                Assert.Equal(2, reader.Take());
-                Assert.Equal(3, reader.Take());
-                Assert.Equal(-1, reader.Take());
-            }
+            var reader = new ReadableBufferReader(ReadableBuffer.Create(new byte[] { 1, 2, 3, 4, 5 }, 0, 5));
+            reader.Skip(5);
+            Assert.True(reader.End);
+            Assert.Equal(-1, reader.Peek());
+            Assert.True(reader.Cursor.IsEnd);
         }
 
         [Fact]
-        public async Task PeekTraversesSegments()
+        public void SkipSingleBufferSkipsBytes()
         {
-            using (var factory = new PipeFactory())
-            {
-                var readerWriter = factory.Create();
-                var w = readerWriter.Writer.Alloc();
-                w.Append(ReadableBuffer.Create(new byte[] { 1 }, 0, 1));
-                w.Append(ReadableBuffer.Create(new byte[] { 2 }, 0, 1));
-                await w.FlushAsync();
-
-                var result = await readerWriter.Reader.ReadAsync();
-                var buffer = result.Buffer;
-                var reader = new ReadableBufferReader(buffer);
-
-                Assert.Equal(1, reader.Take());
-                Assert.Equal(2, reader.Peek());
-                Assert.Equal(2, reader.Take());
-                Assert.Equal(-1, reader.Peek());
-                Assert.Equal(-1, reader.Take());
-            }
+            var reader = new ReadableBufferReader(ReadableBuffer.Create(new byte[] { 1, 2, 3, 4, 5 }, 0, 5));
+            reader.Skip(2);
+            Assert.Equal(3, reader.Peek());
+            reader.Skip(2);
+            Assert.Equal(5, reader.Peek());
         }
 
         [Fact]
-        public async Task PeekWorkesWithEmptySegments()
+        public void SkippingPastLengthThrows()
         {
-            using (var factory = new PipeFactory())
-            {
-                var readerWriter = factory.Create();
-                var w = readerWriter.Writer.Alloc();
-                w.Append(ReadableBuffer.Create(new byte[] { 0 }, 0, 0));
-                w.Append(ReadableBuffer.Create(new byte[] { 1 }, 0, 1));
-                await w.FlushAsync();
-
-                var result = await readerWriter.Reader.ReadAsync();
-                var buffer = result.Buffer;
-                var reader = new ReadableBufferReader(buffer);
-
-                Assert.Equal(1, reader.Peek());
-                Assert.Equal(1, reader.Take());
-                Assert.Equal(-1, reader.Peek());
-                Assert.Equal(-1, reader.Take());
-            }
+            var reader = new ReadableBufferReader(ReadableBuffer.Create(new byte[] { 1, 2, 3, 4, 5 }, 0, 5));
+            Assert.Throws<ArgumentOutOfRangeException>(() => reader.Skip(6));
+        }
+
+        [Fact]
+        public void CtorFindsFirstNonEmptySegment()
+        {
+            var buffer = BufferUtilities.CreateBuffer(new[] { new byte[] { }, new byte[] { 1 } });
```

Comment:
If you don't care about values use `int[]` overload BufferUtilities.CreateBuffer(0, 1)